app/assets/javascripts/discourse/app/components/site-header.js

```diff
@@ -177,6 +177,19 @@ const SiteHeaderComponent = MountWidget.extend(
         this.docAt = header.offsetTop;
       }
 
+      const headerRect = header.getBoundingClientRect();
+      let headerOffsetCalc = headerRect.top + headerRect.height;
+
+      if (window.scrollY < 0) {
+        headerOffsetCalc += window.scrollY;
+      }
+
+      const newValue = `${headerOffsetCalc}px`;
+      if (newValue !== this.currentHeaderOffsetValue) {
+        this.currentHeaderOffsetValue = newValue;
```

Comment:
In https://github.com/discourse/discourse/commit/9a55c9c4339214ea18a73e71ab2506e4ad2967aa, Joffrey noted that this code sometimes would set a negative header-offset value. Did you check that restoring this doesn't restore that issue @techAPJ? 

I suspect it would set negative values on touch devices, where swiping can trigger a scroll overflow.